staging/src/k8s.io/apimachinery/pkg/runtime/extension.go

```diff
@@ -25,7 +26,10 @@ func (re *RawExtension) UnmarshalJSON(in []byte) error {
 	if re == nil {
 		return errors.New("runtime.RawExtension: UnmarshalJSON on nil pointer")
 	}
-	re.Raw = append(re.Raw[0:0], in...)
+
+	if !bytes.Equal(in, []byte("null")) {
+		re.Raw = append(re.Raw[0:0], in...)
```

Comment:
I think making UnmarshalJSON round trip MarshalJSON correctly makes sense...

@smarterclayton @deads2k you have spent some quality time with RawExtension... is there any reason you can think of we'd need to preserve `null` past the json decoding phase?